File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    # yellow의 약수를 구함
    yellow_yaksu = []
    for i in range(1, yellow+1):
        if (yellow%i==0):
            yellow_yaksu.append(i)
    
    
    for i in yellow_yaksu:
        for j in yellow_yaksu:
            if (i<j):
                continue
            if (i>=j and i*j == yellow and ((i+2))*(j+2)==(yellow+brown)):
                answer.append(i+2)
                answer.append(j+2)
                return answer
    
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        List<Integer> yellowYaksu = new ArrayList<>();
        
        for (int i = 1; i <= yellow; i++) {
            if (yellow % i == 0) {
                yellowYaksu.add(i);
            }
        }
        
        for (int i : yellowYaksu) {       
            for (int j : yellowYaksu) { 
                if (i < j) {
                    continue;
                }
                
                if (i * j == yellow && (i + 2) * (j + 2) == (yellow + brown)) {
                    answer[0] = i + 2; 
                    answer[1] = j + 2; 
                    return answer;
                }
            }
        }
        
        return answer;
    }
}